libnmstate/nm/route.py

```diff
@@ -63,14 +64,18 @@ def get_config(acs_and_ip_profiles):
     """
     routes = []
     for (active_connection, ip_profile) in acs_and_ip_profiles:
-        if not ip_profile.props.routes:
+        if not ip_profile.props.routes and not ip_profile.props.gateway:
             continue
         iface_name = _get_iface_name(active_connection)
         if not iface_name:
             raise NmstateInternalError(
                 'Got connection {} has not interface name'.format(
                     active_connection.get_id()))
         default_table_id = ip_profile.props.route_table
+        if ip_profile.props.gateway:
```

Comment:
You already have the gateway value, the only missing value is the route_metric.
How about fetching it at this level and calling the _get_default_route_config with it instead of passing the ip_profile?